packages/application-shell/src/components/setup-flop-flip-provider/setup-flop-flip-provider.tsx

```diff
@@ -18,21 +25,68 @@ type Props = {
   shouldDeferAdapterConfiguration?: boolean;
 };
 
-type CustomLDUser = {
-  project: string;
-  id: string;
-  team: string[];
-  group: string;
-  subgroup: string;
-  tenant: string;
+type TLaunchDarklyUserCustomFields = {
+  project?: string;
+  id?: string;
+  team?: string[];
+  group?: string;
+  subgroup?: string;
+  tenant?: string;
+};
+
+type TFetchedGraphQLAdapterFlag = {
+  name: string;
+  value: boolean;
+  reason?: string;
 };
 
 // This value is hard-coded here because we want to make sure that the
 // app uses our account of LD. The value is meant to be public, so there
 // is no need to be concerned about security.
 const ldClientSideIdProduction = '5979d95f6040390cd07b5e01';
 
+combineAdapters.combine([ldAdapter, httpAdapter]);
+
+function getUserCustomFieldsForLaunchDarklyAdapter(
+  user?: Props['user'],
+  projectKey?: string
+): TLaunchDarklyUserCustomFields {
+  if (!user && !projectKey) {
+    return {};
+  }
+
+  const customFieldsOfUser = {
+    id: user.launchdarklyTrackingId,
+    team: user.launchdarklyTrackingTeam,
+    group: user.launchdarklyTrackingGroup,
+    subgroup: user.launchdarklyTrackingSubgroup,
+    tenant: user.launchdarklyTrackingTenant,
+  }
+
+  if (!project) return customFieldsOfUser
+
+  return {
+    project: projectKey,
+    ...customFieldsOfUser
+  };
```

Comment:
@emmenko this is why I liked the `omitEmpty` also if it was, I admit you caught me, lying about the types. Kind of funny that you can with generics.

How would you like to make this more explicit? TypeScript doesn't like this I think. Thanks.